File: src/features/typing/data/ts/best_time_to_buy_and_sell_stock.ts
function maxProfit(prices: number[]): number {
  let minPrice = prices[0];
  let best = 0;
  for (const price of prices) {
    if (price < minPrice) {
      minPrice = price;
    } else {
      best = Math.max(best, price - minPrice);
    }
  }
  return best;
}

File: src/features/typing/data/dart/best_time_to_buy_and_sell_stock.dart
import 'dart:math';

class Solution {
  int maxProfit(List<int> prices) {
    int minPrice = prices[0];
    int best = 0;
    for (var price in prices) {
      if (price < minPrice) {
        minPrice = price;
      } else {
        best = max(best, price - minPrice);
      }
    }
    return best;
  }
}
